pennylane/transforms/registrations.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the global dictionaries for tranform registrations.
+"""
+
+# Mapping from operation types to methods that create
+# concrete controlled versions of a given operation.
+# Functions should take the given operation as input
+# and return nothing. The desired controlled operation(s)
+# should be added to the tape context by setting do_queue=True.
+CONTROL_MAPS = {}  # Dict[type, Callable[Operation, None]]
+
+
+def register_control(cls, fn):
+    """Register the control transform for a custom Operation.
+
+    TODO(chase): Documentation.
+    """
+    CONTROL_MAPS[cls] = fn

```

Comment:
So this dictionary is filled in each import/run of PennyLane? If a user adds a control replacement, they can do it anywhere in their code and it will be stored for the scope of the run?

Reply:
Yep! This is very similar to how autograd registers their gradient rules. https://github.com/HIPS/autograd/blob/01eacff7a4f12e6f7aebde7c4cb4c1c2633f217d/autograd/core.py#L42